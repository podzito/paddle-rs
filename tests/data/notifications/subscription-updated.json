{
  "data": {
    "address_id": "add_01j5damdb8c4f3pqwszree8sy9",
    "billing_cycle": { "frequency": 1, "interval": "month" },
    "billing_details": null,
    "business_id": null,
    "canceled_at": null,
    "collection_mode": "automatic",
    "created_at": "2024-08-16T10:10:19.701Z",
    "currency_code": "USD",
    "current_billing_period": { "ends_at": "2024-09-16T10:10:18.664529Z", "starts_at": "2024-08-16T10:10:18.664529Z" },
    "custom_data": null,
    "customer_id": "ctm_01j535fawggst1gm3qsyanc0mb",
    "discount": null,
    "first_billed_at": "2024-08-16T10:10:18.664529Z",
    "id": "sub_01j5danadndbm1rwfag40173qe",
    "import_meta": null,
    "items": [
      {
        "created_at": "2024-08-16T10:10:19.701Z",
        "next_billed_at": null,
        "previously_billed_at": "2024-08-16T10:10:18.664529Z",
        "price": {
          "billing_cycle": { "frequency": 1, "interval": "month" },
          "created_at": "2024-08-12T09:40:22.442876Z",
          "custom_data": { "credits": "10", "storage": "500" },
          "description": "PodZito Std - Monthly",
          "id": "pri_01j52zbk9avbcnce2b4n893a60",
          "import_meta": null,
          "name": "PodZito Std",
          "product_id": "pro_01j52z87e5sdt3p60mmtcavpgt",
          "quantity": { "maximum": 1, "minimum": 1 },
          "seller_id": "21860",
          "status": "active",
          "tax_mode": "account_setting",
          "trial_period": null,
          "type": "standard",
          "unit_price": { "amount": "1499", "currency_code": "USD" },
          "unit_price_overrides": [],
          "updated_at": "2024-08-13T17:51:16.988712Z"
        },
        "product": {
          "created_at": "2024-08-12T09:38:32.005Z",
          "custom_data": null,
          "description": "Podzito",
          "id": "pro_01j52z87e5sdt3p60mmtcavpgt",
          "image_url": null,
          "import_meta": null,
          "name": "PodZito",
          "seller_id": "21860",
          "status": "active",
          "tax_category": "standard",
          "type": "standard",
          "updated_at": "2024-08-12T09:38:32.005Z"
        },
        "quantity": 1,
        "recurring": true,
        "status": "active",
        "trial_dates": null,
        "updated_at": "2024-08-16T10:10:19.701Z"
      }
    ],
    "next_billed_at": null,
    "paused_at": null,
    "scheduled_change": { "action": "cancel", "effective_at": "2024-09-16T10:10:18.664529Z", "resume_at": null },
    "started_at": "2024-08-16T10:10:18.664529Z",
    "status": "active",
    "updated_at": "2024-08-16T10:10:43.247Z"
  },
  "event_id": "evt_01j5dap1wevyejk4vnv5vzj5hf",
  "event_type": "subscription.updated",
  "notification_id": "ntf_01j5dap22x8bck7kmexv0j1y60",
  "occurred_at": "2024-08-16T10:10:43.726724Z"
}

