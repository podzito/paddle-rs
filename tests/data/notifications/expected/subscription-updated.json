{
  "event_id": "evt_01j6feqkmc1jkbv5d0js9j0thr",
  "notification_id": "ntf_01j6feqkqytbc215anc54hr32e",
  "event": {
    "SubscriptionUpdated": {
      "billing_cycle": { "frequency": 1, "interval": "month" },
      "next_billed_at": "2024-09-29T16:15:36.182Z",
      "paused_at": null,
      "scheduled_change": null,
      "started_at": "2024-08-29T16:12:10.967Z",
      "items": [
        {
          "price": {
            "billing_cycle": { "frequency": 1, "interval": "month" },
            "created_at": "2024-08-12T09:42:40.160401Z",
            "custom_data": null,
            "description": "PodZito Pro - Monthly",
            "id": "pri_01j52zfss0hcfxamnma1tf42gt",
            "import_meta": null,
            "name": "PodZito Pro",
            "product_id": "pro_01j52z87e5sdt3p60mmtcavpgt",
            "quantity": { "maximum": 1, "minimum": 1 },
            "status": "active",
            "tax_mode": "account_setting",
            "trial_period": { "frequency": 7, "interval": "day" },
            "type": "standard",
            "unit_price": { "amount": "999", "currency_code": "USD" },
            "updated_at": "2024-08-29T11:35:06.922573Z"
          },
          "product": {
            "created_at": "2024-08-12T09:38:32.005Z",
            "custom_data": null,
            "description": "Podzito",
            "id": "pro_01j52z87e5sdt3p60mmtcavpgt",
            "image_url": null,
            "name": "PodZito",
            "status": "active",
            "tax_category": "standard",
            "type": "standard",
            "updated_at": "2024-08-12T09:38:32.005Z"
          },
          "created_at": "2024-08-29T16:12:10.973Z",
          "next_billed_at": "2024-09-29T16:15:36.182Z",
          "previously_billed_at": "2024-08-29T16:15:36.182Z",
          "quantity": 1,
          "recurring": true,
          "status": "active",
          "trial_dates": { "starts_at": "2024-08-29T16:12:10.967Z", "ends_at": "2024-08-29T16:15:36.182Z" },
          "updated_at": "2024-08-29T16:15:36.196Z"
        }
      ],
      "status": "active",
      "transaction_id": null,
      "updated_at": "2024-08-29T16:15:36.195Z"
    }
  },
  "occurred_at": "2024-08-29T16:15:39.660663Z"
}
